packages/main/src/plugin/index.ts

```diff
@@ -120,15 +123,63 @@ export class PluginSystem {
     });
   }
 
+  // Create an Error to access stack trace
+  // Match a regex for the file name and return it
+  // return nothing if file name not found
+  async getExtName(){
+    //Create an error for its stack property
+    const stack = new Error().stack
+    //Create a map for extension path => extension name
+    let extensions: Map<string, string> = new Map();
+
+    //Check if index.ts private member extensionLoader initialized
+    //If it is grab listExtensions()
+    if(this.extensionLoader != undefined){
+      let validExtObj = await this.extensionLoader.listExtensions();
+      //start setting path => name
+      validExtObj.forEach((extInfo: ExtensionInfo) => {
+        extensions.set(extInfo.path, extInfo.name)
+      })
+    }
```

Comment:
as an improvement, I think we should build objects only if extension list is changing.
It'll cost too many iteration to do that on each log call.